### PR TITLE
Added features + fixed issues

### PR DESCRIPTION
**Pull request by Electric131**
Allows players to use /name <uuid> to get the name of a player(by their uuid). (Useful when dealing with vars that store the UUID of a player)

### CODE REVIEW

**Comment by Electric131** [comment]
Still a few things that I am adding but it should be ready to be merged with no more debug logs. (which I forgot one)

**Comment by Electric131** [comment]
The error detection fix only works in certain scenarios, so I will be trying to improve that.

**Comment by tk2217** [comment]
This seems to have ballooned in scope over just the name command, maybe extract some of this into multiple pull requests? Additionally, I think we may have already been working on some bigger changes to some of these systems.

**Comment by Electric131** [comment]
Yeah it became a lot more than just /name command.

**Review by homchom — COMMENTED** [review]
Finally got around to looking at this. It's 11pm for me, so I apologize if I am not looking at everything correctly. Here are my thoughts:

1. What is /name useful for anyway? You have to give it a UUID, which isn't pleasant to type manually, so you would either have to copy it from logs or something (at which point it'd be easier to just change how your plot stores save vars; for example, use numerical ids instead of uuids), or you would have to copy it from the result of another command (like /uuid, at which point its moot). What would you use it for, specifically?
2. If we do end up adding /name, they should probably be consolidated since they're so specific and similar. I'm thinking a generic `/convert <to>` command (eg. `/convert uuid homchom`).
~~3. The change you made to detecting support sessions won't work, because [SUPPORT] messages are only sent to support. This would break for all non-staff :)~~ see below
4. What's with all the new threads and sleep calls for sending locates? What's the rationale behind the specific wait times? Is there not a better way?
5. If I understand correctly, your current iteration of the state detection code does not detect going to spawn. Should we fix this somehow? My only idea right now (beside a /locate timer) is hardcoding the spawns for each node, but that isn't ideal.
6. Could you summarize your changes to code block error detection for me? How does it work now, as opposed to before? What are the benefits?
7. When fixing things like /chat local, did you just add new code to implement them? It looks like you just "added" them; there should be code already that needs to be fixed instead, unless you're saying that it got removed by accident.

EDIT: looking at your changes to state detection more, I see now what you are trying to do with support sessions. So let me ask for clarification (since I don't remember myself either): is "entering a session" referring to a supportee, a supporter, or both? If both, you'll probably need to check for two messages separately (unless there's a common message sent to both players).

**Comment by homchom** [comment]
Update: looking at your changes to state detection more, I see now what you are trying to do with support sessions. So let me ask for clarification (since I don't remember myself either): is "entering a session" referring to a supportee, a supporter, or both? If both, you'll probably need to check for two messages separately (unless there's a common message sent to both players).

**Comment by Electric131** [comment]
Answers to your questions:
1. /name is useful if you have a plot with no conversion and you store records of hundreds of players using only UUIDs and need to find the player it is linked to.
2. Yeah they should be merged :P, this was my first time using Java so making a command seemed easy since I just modified the /uuid code.
3. The code is only for support members, supportees are just left out ig. :P
4. The specific times were from a lot of trial and error, because if you go to a plot cross nodes it can add some delay so I had to change those quite a lot to get it pretty accurate. It also has a failsafe check after about 2s in case the specific time doesn't work.
5. My testing shows that going to spawn does work, if you could show a picture of it failing I can try to find out why.
6. The code block error detection is kind of complicated so it is hard to summarize but I completely reworked the old code and make it work step by step going through the arguments. It uses several checks to find pieces of different options (parts with OR) and checking if the type of the item in the slot matches any. It is hard to summarize beyond that.
7. I don't recall fixing the /char local, but I moved all the old auto commands into my new state detection so I did change the **auto** chat local.

**Comment by homchom** [comment]
This is impressive for your first time using Java!

2. Can you do that, or do you want me to?
3. The old messages don't seem "wrong" though; they are messages sent to the supportee (but maybe not the staff). Either way, maybe we should reconsider state detection about sessions, as ideally they should be supported always or not at all.
4. This is not a good way to deal with that. Instead, you should run the code once you know it is safe to do so. It of course depends on the specific case (eg. entering a session vs. going to spawn), but there is probably an event you can listen to that confirms everything is loaded, and then we can design a system to only send the locate once all checks are met. If you'd like, I can do that and commit to this PR.
5. I haven't tested it; I'm just not sure why it would. Where in the code is spawn detection done?
6. I'm mainly worried about performance, but if you're saying it works, I can look at it more later before merging dev to main.
7. Sorry, I misspoke. I meant auto /chat local. And thanks, that answers my question.

**Comment by Electric131** [comment]
2. It would be better if you could do it
3. I didn't know the old messages were for supportees however supporting both is a good idea
4. I would love to see any idea for a confirmation event to check as my time checks are a bad method right now and they even use the fallback sometimes
5. The spawn detection code is on inventory update to check your slot item, which also uses the time system to locate them iirc
6. Yeah you can go ahead and test it but there don't seem to be any noticeable performance drops and the only issue is sometimes the gui doesn't load in at all.

**Comment by homchom** [comment]
Alright. I will work on these and add them before the next beta releases. In the future, please keep unrelated additions/fixes/changes to separate pull requests, so I don't have to merge unfinished code into a branch that's supposed to be stable, or especially into other feature branches.
